docs/api/field_type_search.md

```diff
@@ -1,18 +1,17 @@
-# Search
+# Field Type searching
 
-Fields, or a custom Field Type, might contain or maintain data relevant for user searches. To make the search engine aware of the data in your Field Type you need to implement an additional interface and register the implementation.
+Fields, or a custom Field Type, might contain or maintain data relevant for user searches.
+To make the search engine aware of the data in your Field Type you need to implement an additional interface and register the implementation.
 
-If your Field Type does not maintain any data, which should be available to search engines, feel free to just ignore this section.
+The `eZ\Publish\SPI\FieldType\Indexable` interface defines the methods below which are required if the Field Type provides data relevant to search engines.
 
-The `eZ\Publish\SPI\FieldType\Indexable` defines below methods, which are required to be implemented, if the Field Type provides data relevant to search engines.
-
-`getIndexData( Field $field )`
+### `getIndexData( Field $field )`
 
 This method returns the actual index data for the provided `eZ\Publish\SPI\Persistence\Content\Field`. The index data consists of an array of `eZ\Publish\SPI\Persistence\Content\Search\Field` instances. They are described below in further detail.
 
-`getIndexDefinition()`
+### `getIndexDefinition()`
 
-To be able to query data properly an indexable Field Type also is required to return search specification. You must return a HashMap of `eZ\Publish\SPI\Persistence\Content\Search\FieldType` instances from this method, which could look like:
+To be able to query data properly an indexable Field Type also is required to return search specification. You must return an array of `eZ\Publish\SPI\Persistence\Content\Search\FieldType` instances from this method, which could look like:
```

Comment:
any particular reason for replacing HashMap with an array? HashMap is a specific case of an array, which is here relevant. More understandable term here would be "an associative array".